File: core/src/main/java/co/ericp/freewayforecast/routes/Leg.java
package co.ericp.freewayforecast.routes;

import java.util.List;

import co.ericp.freewayforecast.Location;

/**
 * One leg of a trip. Trips without waypoints only have one leg.
 */
public class Leg {
    double distance;
    long duration;

    Location start;
    Location end;

    List<Step> steps;
}

File: core/src/main/java/co/ericp/freewayforecast/routes/Leg.kt
package co.ericp.freewayforecast.routes

import co.ericp.freewayforecast.Location

/**
 * One leg of a trip. Trips without waypoints only have one leg.
 */
data class Leg(
        val start: Location,
        val end: Location,
        val distance: Long,
        val duration: Long,
        val steps: List<Step>
)
